### DISCUSSION TITLE
[V5] Lucid query.with() throwing error

### DISCUSSION THREAD

**Discussion by dievardump** [adonisjs/core#2203] [Help]
When using `Model.query().with('relation')` I get the error

`"with() second argument must be a function / QueryBuilder or a raw"` although the relationship is declared on my Model with a 

```ts
  @column()
  public relationId: number;

  @belongsTo(() => Relation)
  public relation: BelongsTo<typeof Relation>;
```

What is the expected usage here? Shouldn't Lucid find the relation automatically depending on the BelongsTo?

**Comment by McSneaky** [comment]
Use `.preload('relation')`?

**Reply by dievardump** [reply]
I want to only select a few columns of `relation` and I've not really been able to do that using `preload`.

When logging `query.toSQL()` preload doesn't seem to add to the SQL Query. So If preload means that queries are executed on top of the first one, I would prefer do everything with one.

**Reply by McSneaky** [reply]
If you want to select only certain fields then 
```js
.preload('relation', (relationQuery) => {
  relationQuery.select('id', 'name', 'foo', 'bar')
})
```
If you want to log SQL queries, then you can add debug:true to your DB config or add `DEBUG=knex:*` env variable. If that's too verbose, then you can limit debug scope with something like `DEBUG=knex:query`

But if I recall correctly .preload() did additional query. Might remember it wrong tho

**Reply by dievardump** [reply]
Thank you for replying.

If preload does SQL queries on top, then I would really like to go with `with` and understand what it expects from me although the relation is defined (withCount seems to have no problem seeing relations automatically). 

~~Concerning debug: I've used both `config.database.pg.debug: true;`  and `DEBUG=knex:query` but no SQL logs to be found~~ nevermind, I needed to add event listener
